source/MaterialXCore/Definition.cpp

```diff
@@ -37,6 +38,44 @@ InterfaceElementPtr NodeDef::getImplementation(const string& target) const
     return InterfaceElementPtr();
 }
 
+InterfaceElementPtr NodeDef::getImplementation(const string& language, const string& target) const
+{
+    for (InterfaceElementPtr element : getDocument()->getMatchingImplementations(getName()))
+    {
+        // Check direct implementation
+        ImplementationPtr implementation = element->asA<Implementation>();
+        if (implementation)
+        {
+            // Check for a language match
+            const std::string& lang = implementation->getLanguage();
+            if (lang.length() && lang == language)
+            {
+                // Check target. Note that it may be empty.
+                if (targetStringsMatch(implementation->getTarget(), target))
+                {
+                    return implementation;
+                }
+            }
+        }
+        // Check for a nodegraph implementation which does not require a
+        // language check
+        else if (element->isA<NodeGraph>())
+        {
+            if (targetStringsMatch(implementation->getTarget(), target))
+            {
+                return element;
+            }
+        }
+    }
+    return InterfaceElementPtr();
+}
+
+bool NodeDef::requiresImplementation() const
```

Comment:
This may turn out to be the right solution in the future, but I'd like to hold off on declaring a NodeDef::requiresImplementation method just yet.  Once we've decided upon how to handle "unimplementable" nodes like ambient occlusion, then we can revisit this and write up a universal definition.